### PR TITLE
jewel: osd: omap threadpool heartbeat is only reset every 100 values

### PR DESCRIPTION
**Pull request by Vicente-Cheng**
http://tracker.ceph.com/issues/20492

### CODE REVIEW

**Comment by Vicente-Cheng** [comment]
retest this please

File: src/osd/ReplicatedBackend.cc [hunk 1]
```diff
@@ -804,13 +804,8 @@ void ReplicatedBackend::be_deep_scrub(
...
       poid, ghobject_t::NO_GEN, get_parent()->whoami_shard().shard));
   assert(iter);
-  uint64_t keys_scanned = 0;
   for (iter->seek_to_first(); iter->valid() ; iter->next(false)) {
-    if (cct->_conf->osd_scan_list_ping_tp_interval &&
-	(keys_scanned % cct->_conf->osd_scan_list_ping_tp_interval == 0)) {
-      handle.reset_tp_timeout();
-    }
-    ++keys_scanned;
+    handle.reset_tp_timeout();
```
Comment (liu-chunmei) [inline src/osd/ReplicatedBackend.cc]: what's the benefit we reset timeout every each iter?

Comment (Vicente-Cheng) [inline src/osd/ReplicatedBackend.cc]: hi @liu-chunmei , sorry for late reply
I think the origin pull request have a explain. #15823
In some cases, reset thread heartbeat every 100 entries would get more cost than every omap.

**Comment by smithfarm** [comment]
@jdurgin @tchaikov This passed a rados run at http://tracker.ceph.com/issues/20613#note-40 with some failures, all of which I tried to reproduce. The only one that is reliably reproducible is http://tracker.ceph.com/issues/20981

Please review.

**Pull request merged**